includes/admin/class-wc-rest-stripe-account-controller.php

```diff
@@ -98,13 +106,22 @@ public function get_account() {
 	public function get_account_summary() {
 		$account = $this->account->get_cached_account_data();
 
+		// Use statement descriptor from settings, falling back to Stripe account statement descriptor if needed.
+		$statement_descriptor = WC_Stripe_Helper::clean_statement_descriptor( $this->gateway->get_option( 'statement_descriptor' ) );
+		if ( empty( $statement_descriptor ) ) {
+			$statement_descriptor = $account['settings']['payments']['statement_descriptor'];
+		}
+		if ( empty( $statement_descriptor ) ) {
+			$statement_descriptor = WC_Stripe_Helper::clean_statement_descriptor( get_bloginfo( 'name' ) );
+		}
+
 		return new WP_REST_Response(
 			[
 				'has_pending_requirements' => $this->account->has_pending_requirements(),
 				'has_overdue_requirements' => $this->account->has_overdue_requirements(),
 				'current_deadline'         => $account['requirements']['current_deadline'] ?? null,
 				'status'                   => $this->account->get_account_status(),
-				'statement_descriptor'     => $account['settings']['payments']['statement_descriptor'] ?? '',
+				'statement_descriptor'     => $statement_descriptor,
```

Comment:
Should we make these defaults consistent in other places where we use the statement descriptor? Like in[ the gateways](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php#L131), for example.

Thinking on a scenario where a customer would make a purchase on the same store through the site and then through the app, and the statements descriptors would be different for each purchase.

Not sure if this would be the case or if this is something we'd like to adjust. I don't think this is a blocker either way. We could check this out in a different issue if it's something we'd like to address.

Reply:
It looks like the way the extension currently behaves is to pass `null` to Stripe if there is no statement descriptor, and Stripe then uses a default value of "Stripe" as the descriptor if there's none present.

Let me check with @allendav to see what the ideal behavior is from mobile's perspective. If this API returned `null` for the statement descriptor if none is present, could the apps pass on the `null` value to the payment intents API to let Stripe use its own default statement descriptor? (It seems to me from the Stripe's API docs that `null` is allowed by the empty string is not.) Or would there be some other reason that would require a non-null value?

Reply:
I'll take a look

Reply:
It works and although not explicitly stated in the text, the optional type (`String?`) suggests it in the docs as well: 

https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPPaymentIntentParameters.html#/c:objc(cs)SCPPaymentIntentParameters(py)statementDescriptor

I'll post a PR to the mobile app to pass nil to Stripe when creating a payment intent for an in-person payment when an empty string is given by the account endpoint

fyi @ctarda @malinajirka

Reply:
Thanks for checking, @allendav! If Stripe's API accepts nil, I'm inclined to have the `wc_stripe/account/summary` endpoint in the extension return `null` for the statement descriptor (instead of the empty string), if that works for you?

Reply:
I'll test that today and see if that causes any iOS hiccups. cc @malinajirka for Android

Reply:
We'll need to update the Android app as `statement_descriptor` is currently defined as a non-nullable String. Having said that, it shouldn't cause any issues. Just let me know when you make the final decision and I'll update the code accordingly. (just a note, that we can't do that for WCPay as it's a breaking change - Stripe extension hasn't been released yet, so it should be safe)

Reply:
@allendav @malinajirka I've updated this branch to:

- Return the WC Stripe settings statement descriptor if non-empty.
- Otherwise, return the Stripe settings statement descriptor if non-empty.
- Otherwise, return `null`.

As a result, you should be able to pass what you get from this API directly to Stripe on payment intent creation, since it will always be either a non-empty string or null. Can you test this on iOS/Android to see if this behavior will work for you?

Reply:
Can do

Reply:
Thanks @brianyu28!

I've tested the following settings of the statement descriptor(on Android):
1. Extension (empty), Stripe (empty) - 🟢 it's automatically set to "Stripe"
2. Extension ("my statement descriptor"), Stripe (empty) - 🟢 "my statement descriptor" was used
3. Extension (empty), Stripe ("my statement descriptor) - 🟠  null was returned from the API, is there perhaps some cache? 

To run the test `3` I first emptied the value in the extension and I updated the descriptor in Stripe's dashboard afterwards. 
However, if I first update the value in Stripe's dashboard and then update the value in the extension, it is correctly set to whatever is set on Stripe's dashboard. So I think this PR works as expected, the extension just fetches the current value from Stripe only when the statement descriptor field is updated in the extension (or perhaps after some time)

Reply:
The extension does cache Stripe account data so it's not hitting Stripe's API for every request, so this sounds like the expected behavior. Thanks for testing, @malinajirka! Planning to merge this today so that it's included in the 6.2.0 release.